packages/flutter/lib/src/painting/text_style.dart

```diff
@@ -230,10 +232,13 @@ class TextStyle extends Diagnosticable {
     this.decoration,
     this.decorationColor,
     this.decorationStyle,
+    String debugLabel,
     String fontFamily,
     String package,
   }) : fontFamily = package == null ? fontFamily : 'packages/$package/$fontFamily',
-       assert(inherit != null);
+       debugLabel = debugLabel == null ? _kDefaultDebugLabel : debugLabel,
```

Comment:
`debugLabel = debugLabel ?? _kDefaultDebugLabel`

Or, if it doesn't interfere with other logic, just make it a default and assert non-null.

Reply:
Alternatively, default to null, which will be more efficient in release builds and in comparisons against the default value, and only bringing _kDefaultDebugLabel when you need to merge strings (using `debugLabel ?? _kDefaultDebugLabel`)